docs/upgrade.html

```diff
@@ -145,7 +145,7 @@ <h5><a id="upgrade_350_kraft" href="#upgrade_350_kraft">Upgrading KRaft-based cl
             </code>
         </li>
         <li>Note that the cluster metadata version cannot be downgraded to a pre-production 3.0.x, 3.1.x, or 3.2.x version once it has been upgraded.
-            However, it is possible to downgrade to production versions such as 3.3-IV0, 3.3-IV1, etc.</li>
+            However, it is possible to downgrade to a backwards compatible metadata version. The active Controller will block any incompatible downgrade attempt.</li>
```

Comment:
From what I can see:
https://github.com/apache/kafka/blob/fbc39bcdbe5ebcb07435f467b77f4b7df2e910f4/server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java#L170-L186

The version v3.6.0 (using `IBP_3_6_IV2`) and v3.5.0 (using `IBP_3_5_IV2`) and older ones are not able to downgrade to any version since the last parameter (`didMetadataChange`) is true, right? If so, the line is still confusing users, right?

Reply:
I'm thinking we can direct users to check the MetadataVersion.java file to see if they are backward compatible. WDYT?

Reply:
I thought about that, but doesn't seem a great user experience. That said, I don't see a better way, unless we want to create a new tool method to verify if downgrade is possible between any two versions, and document it.

Reply:
In `features.sh`, there's a `dry_run` option to test if it is able to downgrade. For this doc update, I think we just need to make it clear without confusion.

Reply:
Yes, but this doesn't help you if you want to verify if a downgrade is possible before upgrading to a new version. In that case, you would need to create a test cluster.

Reply:
Make sense. We can create a KIP for that. But for this PR, I still think our goal is to make the doc clear.